pkg/build/controller/controller.go

```diff
@@ -125,29 +125,8 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	// Set the build phase, which will be persisted if no error occurs.
 	build.Status.Phase = buildapi.BuildPhasePending
 
-	// Make a copy to avoid mutating the build from this point on.
```

Comment:
@liggitt it is not ok to mutate the build (spec), this didn't change.
This code here was a hack -- since we cannot mutate the spec, we were making a copy and mutating the copy.
@PI-Victor is cleaning up after #4780